src/pages/UserProfilePage/UserProfilePage.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useEffect, useState } from "react";
+import { useParams } from "react-router";
+import { SpecifiedUser } from "../../interfaces/UserInterfaces";
+import useTokenStatus from "../../utils/useTokenStatus";
+import userService from "../../services/userService";
+import { Box, Button, Grid, List, ListItem, Typography } from "@mui/material";
+import UserProfileHeader from "../../components/UserProfile/UserProfileHeader";
+import Loader from "../../components/Shared/Loader";
+import UserProfileHeaderContent from "../../components/UserProfile/UserProfileHeaderContent";
+import FormInputText from "../../components/Shared/FormComponents/FormInputText";
+import { useForm } from "react-hook-form";
+import { ApiErrors } from "../../enums/ErrorsEnums";
+import { useNavigate } from "react-router-dom";
+
+interface IFormInput {
+  note: string;
+}
+
+const UserProfilePage: React.FC = () => {
+  const tokenStatus = useTokenStatus(),
+    id: number = useParams().id as unknown as number,
+    [user, setUser] = useState<SpecifiedUser>(),
+    navigate = useNavigate();
+
+  useEffect(() => {
+    const loadUser = async () => {
```

Comment:
Why not use Suspense for this? That's why we upgraded to React 18

Reply:
Unfortunately, from what I understand, simply replacing the current logic by using Suspense isn't that straight-forward. As far as I can tell, Suspense isn't capable of monitoring the pending promises in your component and as they are pending display the fallback. Instead it gets to the fallback when errors are triggered and in order to properly use it with data fetching components you must wrap the data fetching processes in order to throw errors if the data isn't ready. Or at least this is what I understood after reading https://blog.logrocket.com/react-suspense-data-fetching/.